packages/server/src/controllers/UsersController.js

```diff
@@ -1,172 +1,157 @@
-import Models from '../models'
 import HttpError from '../errors/http-error'
 import UsersValidator from '../validators/UsersValidator'
-
-const { User, UserAddress } = Models
-
-// temp dummy models
-let Users = { }
-let UserEmails = { }
-let UserSessions = { }
+import Users from '../models/objection/Users'
+import UserEmails from '../models/objection/UserEmails'
 
 
 export default {
   async details(req, res) {
     const { params: { address } } = req
-    const body = {
-      emailExists: false,
-      emailVerified: false,
-      addressVerified: false,
-      notificationsDisabled: false,
-      ... Users[address]
-    }
-    res.send(body)
+    const user = await Users.query().findOne({address}).withGraphFetched('[email, emailVerificationToken, notificationSettings]')
+    res.send({
+      emailExists: !!user?.email,
+      emailVerified: !!user?.email && !user?.emailVerificationToken && !!user?.addressVerified,
+      addressVerified: !!user?.addressVerified,
+      notificationsDisabled: !!user?.notificationSettings?.notificationsDisabled
+    })
+  },
+
+
+  async create(req, res) {
+    const params = req.body
+    const errors = await UsersValidator.validateForCreate(params)
+    if (errors.length > 0) throw HttpError._400({ errors })
+    const { email, address } = params
+    let userEmail = await UserEmails.query().findOne({email})
+    if (!userEmail) userEmail= await UserEmails.query().insert({email})
+    await userEmail.$relatedQuery('users').insert({address})
```

Comment:
This is another candidate:

```js
async associateAddress(address) {
   await this.$relatedQuery('users').insert({address})
}
```

Reply:
I don't know, as long as the query is one line it doesn't bother me much. If you insist you can add some, see my other examples :)